src/linux/system.rs

```diff
@@ -507,6 +507,17 @@ impl SystemExt for System {
         get_system_info("NAME=")
     }
 
+    fn get_host_name(&self) -> Option<String> {
+        let hostname_max = unsafe { sysconf(_SC_HOST_NAME_MAX) };
+        let mut buffer = vec![0_u8; (hostname_max as usize) + 1];
+        if unsafe { libc::gethostname(buffer.as_mut_ptr() as *mut c_char, buffer.len()) } == 0 {
```

Comment:
I think you need to shrink the buffer first to account for the NUL byte that `gethostname` terminates its string with otherwise you end up with
```
"ubuntu-host-1\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}\u{0}"
```

Reply:
This is the approach the standard library takes with `CStr`: 
```rust
let null_position = memchr::memchr(0, &buffer)?;
buffer.resize(null_position, 0);
```

Reply:
Oh nice thanks! I'll add that as soon as I work with clippy + Windows XD

Reply:
If you want to not use `memchr`, you can simply iterate over the pointer.  I did it in a few places.

Reply:
I'll leave it up to you as it will be a dependency until `slice_internal` is stabilized. I have it right now using `memchr`.